src/utils/redux/selectEntityTypes.js

```diff
@@ -0,0 +1,47 @@
+/*
+ * @flow
+ */
+
+import {
+  Map,
+  Set,
+  getIn,
+  isCollection,
+} from 'immutable';
+import type { EntityType } from 'lattice';
+
+import { EDM, ENTITY_TYPES, ENTITY_TYPES_INDEX_MAP } from '../../constants/redux';
+import { isNonEmptyArray, isNonEmptyString } from '../lang';
+import { isValidUUID } from '../validation';
+import type { UUID } from '../../types';
+
+export default function selectEntityTypes(idsOrNames :Set<UUID | string> | Array<UUID | string>) {
+
+  return (state :Map) :Map<UUID, EntityType> => {
+
+    const isValid = (
+      (isNonEmptyArray(idsOrNames) || isCollection(idsOrNames))
```

Comment:
I think `Set.isSet(idsOrNames` is more suitable in this case than `isCollection(idsOrNames)`

Reply:
I agree to an extent. Unfortunately, we're not enforcing `immutable` types and additionally, the `:List`, `:Map`, `:Set` annotations are treated as `any` by flow. With that inn mind, I think it can be pretty easy to pass along the incorrect type to these helper functions... like passing in a `List` instead of a `Set`.

As such, using `isCollection` tells me that I'm dealing with an `immutable` object, so even if a `List` was passed instead of a `Set`, the function can still operate correctly.

Reply:
Makes sense. However, I feel that the function should return a regular javascript object if the param is a regular javascript array and an immutable Map otherwise. Sounds like a reasonable expectation from the caller. thoughts?

Reply:
Makes sense. However, I think the function should return a regular javascript object if the param is a regular javascript array and an immutable Map otherwise. Sounds like a reasonable expectation from the caller. thoughts?

Reply:
I don't think it's a good idea to mix return types. We've collectively decided to use `immutable` for redux things, so I strongly feel that a selector should be returning an `immutable` object. It also gives us a great API, like `.isEmpty()` and `.hashCode()`.